packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1063,44 +1210,36 @@ export class UmbrellaStore<M extends BaseMetadata> {
    */
   public deleteInboxNotification(
     inboxNotificationId: string,
-    optimisticUpdateId: string
+    optimisticId: string
   ): void {
-    // Batch 1️⃣ + 2️⃣
     batch(() => {
-      this.removeOptimisticUpdate(optimisticUpdateId); // 1️⃣
-
-      // 2️⃣
-      this.#updateInboxNotificationsCache((cache) => {
-        // Delete it
-        const { [inboxNotificationId]: removed, ...newCache } = cache;
-        return removed === undefined ? cache : newCache;
-      });
+      this.optimisticUpdates.remove(optimisticId);
+      this.notifications.delete(inboxNotificationId);
     });
   }
 
   /**
    * Deletes *all* inbox notifications, replacing the corresponding optimistic
    * update.
    */
-  public deleteAllInboxNotifications(optimisticUpdateId: string): void {
-    // Batch 1️⃣ + 2️⃣
+  public deleteAllInboxNotifications(optimisticId: string): void {
     batch(() => {
-      this.removeOptimisticUpdate(optimisticUpdateId); // 1️⃣
-      this.#updateInboxNotificationsCache(() => ({})); // 2️⃣ empty the cache
+      this.optimisticUpdates.remove(optimisticId);
+      this.notifications.clear();
     });
   }
 
   /**
    * Creates an new thread, replacing the corresponding optimistic update.
    */
   public createThread(
-    optimisticUpdateId: string,
+    optimisticId: string,
     thread: Readonly<ThreadDataWithDeleteInfo<M>>
   ): void {
     // Batch 1️⃣ + 2️⃣
     batch(() => {
-      this.removeOptimisticUpdate(optimisticUpdateId); // 1️⃣j
-      this.#mutateThreadsDB((db) => db.upsert(thread)); // 2️⃣
+      this.optimisticUpdates.remove(optimisticId); // 1️⃣j
```

Comment:
It was here before but:
```suggestion
      this.optimisticUpdates.remove(optimisticId); // 1️⃣
```

Reply:
Thanks, I'll just remove those 1️⃣ and 2️⃣ labels now, since the code has become so small it's easy to see at a glance.